perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -271,81 +262,238 @@ def PercentileCalculator(numbers):
   return result
 
 
-def _ListObjects(storage_schema, bucket, prefix, host_to_connect=None):
-  """List objects under a bucket given a prefix.
+# ### Backends ###
 
-  Args:
-    storage_schema: The address schema identifying a storage. e.g., "gs"
-    bucket: Name of the bucket.
-    prefix: A prefix to list from.
-    host_to_connect: An optional endpoint string to connect to.
+class ObjectStorageBackend(object):
```

Comment:
Eventually, I'd like to move the backend implementation classes to the provider directories, and this one to its own separate file somewhere. How does that sound?

The reason I didn't do it in this script is exactly what you said, that it would require changing how the object storage benchmarking script provisions scripts, but I'm happy to make that part of this PR.